code/modules/projectiles/gun.dm

```diff
@@ -220,7 +223,12 @@ obj/item/weapon/gun/proc/newshot()
 				if( i>1 && !(src in get_both_hands(user))) //for burst firing
 					break
 			if(chambered)
-				if(!chambered.fire(target, user, params, , suppressed, zone_override))
+				var/sprd = 0
+				if(randomspread)
+					sprd = round((rand() - 0.5) * spread)
+				else //Smart spread
+					sprd = round((i / burst_size - 0.5) * spread)
+				if(!chambered.fire(target, user, params, suppressed, zone_override, sprd))
```

Comment:
you removed a "," before suppressed. This will fuck up the order of the argument....